contracts/BlendSwap.sol

```diff
@@ -49,7 +52,7 @@ contract BlendSwap {
         public
     {
         require(
-            status[secretHash] == Status.NOT_INITIALIZED,
+            swaps[secretHash].from == address(0),
```

Comment:
Ok, I see why you do something like this. But wouldn't just `swaps[secretHash] == address(0)` work. Source: https://stackoverflow.com/questions/59448336/check-that-object-is-null-in-solidity-mapping

Reply:
`swaps[secretHash] == address(0)` is ugly, tbh. In the likned answer, they compare an address with an address, which is sane, and you propose to compare a _struct_ with an address, hoping that Solidity will silently coerce a zero address to a struct.

IMO the code is fine, though I'd extract this check into a function with a meaningful name. Alternatively, you can do `swaps[secretHash] == Swap( /*all fields = 0*/ )` but it would likely consume more gas.

Reply:
Nope, I just misinterpreted SO answer, missed an important detail.
> So you have to compare the value to the 0 value for _your_ type